### ISSUE TITLE
A rule to enforce `ofType(NON_STRING)`

### ISSUE THREAD

**Issue by rafaelss95** [timdeschryver/eslint-plugin-ngrx#239]
## Information about the new rule
As mentioned in https://github.com/timdeschryver/eslint-plugin-ngrx/pull/211#discussion_r711401470, we could create a rule for this.

Do you have any name in mind?
- `no-of-type-literal`?
- `use-actions-in-of-type`? (taking the idea of `use-selector-in-select`)
- `prefer-actions-in-of-type`?
- other...

## A snippet of incorrect code
```ts
this.actions$.pipe(ofType('myAction'));
```

## A snippet of correct code
```ts
this.actions$.pipe(ofType(myAction));
```
## I would be willing to submit a PR for the docs :heart:

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by rafaelss95** [comment]
Worth mentioning that this rule is simple achievable using `no-restricted-syntax`:

```json
{
    "rules": {
        "no-restricted-syntax": [
            "error",
            {
                "selector": "CallExpression[callee.name='ofType'] Literal",
                "message": "Prefer using `actions` over `strings` in `ofType`."
            },
        ]
    }
}
```

... or am I missing something?

**Comment by timdeschryver** [comment]
That should do the trick, but this also means that:

- we can't verify that `ofType` is from NgRx
- we can't create a docs page (I think)

If we would create the rule, my vote goes to `use-actions-in-of-type` to keep it consistent

**Comment by rafaelss95** [comment]
> * we can't verify that `ofType` is from NgRx

Hmm, is this really a problem? I mean, for example, `createEffect` we don't verify this either.

> * we can't create a docs page (I think)

Yes, that could be a con.

> If we would create the rule, my vote goes to `use-action-in-of-type` (singular action) to keep it consistent

✅
